File: Java/28.实现 strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        for (int i = 0; i + needle.length() <= haystack.length(); ++i) {
            Boolean match = true;
            for (int j = 0; j < needle.length(); ++j) {
                if (i + j == haystack.length()) {
                    return -1;
                }
                if (haystack.charAt(i+j) != needle.charAt(j)) {
                    match = false;
                    break;
                }
            }
            if (match) {
                return i;
            }
        }
        return -1;
    }
}

File: Golang/28.实现 strStr().go
package leetcode

func strStr(haystack string, needle string) int {
	for i := 0; i+len(needle) <= len(haystack); i++ {
		match := true
		for j := 0; j < len(needle); j++ {
			if i+j == len(haystack) {
				return -1
			}
			if haystack[i+j] != needle[j] {
				match = false
				break
			}
		}
		if match {
			return i
		}
	}
	return -1
}
